### PR TITLE
Add Google TTS

### PR DESCRIPTION
**Pull request by markbackman**
Adding Google's TTS (using RPC API), using SSML for text input to add more control options.

Audio was clicking/popping when streamed. I had to fade-in over 10ms)= to eliminate the popping sound. I think this is an OK solution as it doesn't affect the spoken audio at all.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,9 +9,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+<<<<<<< HEAD
 - Added AWS Polly TTS support and `07m-interruptible-aws.py` as an example.
 
 - Added InputParams to Azure TTS service.
+=======
+- Added Google TTS service and corresponding foundational example `07n-interruptible-google.py`
+>>>>>>> 5fd83d8 (Add Google TTS)
```
Comment (aconchillo) [inline CHANGELOG.md:18]: remove conflict lines.

File: src/pipecat/services/google.py [hunk 2]
```diff
@@ -137,3 +144,195 @@ async def process_frame(self, frame: Frame, direction: FrameDirection):
...
+                    break
+                await self.stop_ttfb_metrics()
+                frame = TTSAudioRawFrame(chunk, self.sample_rate, 1)
+                yield frame
+                await asyncio.sleep(0)  # Allow other tasks to run
+
+            await self.push_frame(TTSStoppedFrame())
+
+        except Exception as e:
+            logger.exception(f"{self} error generating TTS: {e}")
```
Comment (aconchillo) [inline src/pipecat/services/google.py:338]: should we yield an ErrorFrame?

File: src/pipecat/services/google.py [hunk 3]
```diff
@@ -137,3 +144,195 @@ async def process_frame(self, frame: Frame, direction: FrameDirection):
...
+
+            # Convert the response to a mutable numpy array
+            audio_content = np.frombuffer(response.audio_content, dtype=np.int16).copy()
+
+            # Apply a smooth, short fade-in
+            fade_duration = int(0.01 * self.sample_rate)  # 10ms fade-in
+            fade_in = np.square(
+                np.linspace(0, 1, fade_duration)
+            )  # Quadratic fade for smoother start
+            audio_content[:fade_duration] = audio_content[:fade_duration] * fade_in
```
Comment (aconchillo) [inline src/pipecat/services/google.py:322]: Maybe the reason for this is that they send WAV format. You can skip those bytes if that's the case. So, hopefully it's this and we can skip the fade-in part.

```
audio_content = audio_content[44:]
```

File: CHANGELOG.md (hunk 1 above)
Comment (markbackman) [reply CHANGELOG.md]: Oops. Not sure how this got pushed 🤔

File: src/pipecat/services/google.py (hunk 3 above)
Comment (markbackman) [reply src/pipecat/services/google.py]: That did the trick. Thank you!

**Review by aconchillo — APPROVED** [review]

**Comment by aconchillo** [comment]
LGTM!

**Pull request merged**